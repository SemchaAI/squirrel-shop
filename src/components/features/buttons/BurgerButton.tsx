interface IProps {
  isOpen: boolean;
  onClick?: () => void;
}

export const BurgerButton = ({ isOpen, onClick }: IProps) => {
  return (
    <button
      className={`relative z-11 flex h-6 w-8 cursor-pointer flex-col justify-between transition-transform duration-400 md:hidden ${isOpen ? "rotate-y-180" : ""}`}
      onClick={onClick}
      aria-label="Toggle menu"
    >
      <span
        className={`block h-1 rounded bg-high-contrast transition-transform duration-400 ease-in-out ${
          isOpen ? "translate-y-2.5 rotate-45 bg-white" : ""
        }`}
      />
      <span
        className={`block h-1 rounded bg-high-contrast transition-transform duration-400 ease-in-out ${
          isOpen ? "scale-x-0" : ""
        }`}
      />
      <span
        className={`block h-1 rounded bg-high-contrast transition-transform duration-400 ease-in-out ${
          isOpen ? "-translate-y-2.5 -rotate-45 bg-white" : ""
        }`}
      />
    </button>
  );
};
