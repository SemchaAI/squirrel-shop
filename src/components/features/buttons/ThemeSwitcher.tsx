"use client";
import { MoonIcon, SunIcon } from "lucide-react";
import { useTheme } from "next-themes";
import { useEffect, useState } from "react";

export const ThemeSwitcher = () => {
  const [mounted, setMounted] = useState(false);
  const { resolvedTheme, setTheme } = useTheme();
  const nextTheme = resolvedTheme === "light" ? "dark" : "light";
  useEffect(() => setMounted(true), []);

  const toggleTheme = () => {
    setTheme(nextTheme);
  };
  if (!mounted) return <MoonIcon size={32} />;

  return (
    <button
      onClick={toggleTheme}
      aria-label={`Switch to ${nextTheme} mode`}
      title="Toggle light/dark mode"
      className="flex cursor-pointer items-center justify-center"
    >
      {resolvedTheme === "light" ? (
        <MoonIcon role="img" aria-label="Moon" size={32} />
      ) : (
        <SunIcon role="img" aria-label="Sun" size={32} />
      )}
    </button>
  );
};
