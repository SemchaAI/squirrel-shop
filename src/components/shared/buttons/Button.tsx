import { forwardRef } from "react";
import { Loader2 } from "lucide-react";
import clsx from "clsx";

import type { ButtonHTMLAttributes } from "react";

type Variant = "primary" | "ghost" | "text" | "destructive";
type Size = "sm" | "md" | "lg" | "none";
type Rounded = "none" | "md" | "full";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: Variant;
  size?: Size;
  rounded?: Rounded;
  loading?: boolean;
}

const base =
  "inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none";

const variants: Record<Variant, string> = {
  primary: "bg-primary border-primary text-on-primary hover:bg-primary-hover",
  // secondary:
  //   "bg-high-contrast border-high-contrast  text-app hover:bg-on-primary",
  ghost:
    "bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary",
  text: "bg-transparent border-transparent text-primary hover:underline",
  destructive: "bg-error border-error text-white hover:bg-error-hover",
};

const sizes: Record<Size, string> = {
  none: "text-sm",
  sm: "px-3 py-1.5 text-sm",
  md: "px-4 py-2 text-base",
  lg: "px-6 py-3 text-lg",
};

const round: Record<Rounded, string> = {
  full: "rounded-full",
  md: "rounded-md",
  none: "rounded-none",
};

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      children,
      className,
      variant = "primary",
      size = "md",
      loading,
      rounded = "md",
      disabled,
      ...props
    },
    ref,
  ) => {
    return (
      <button
        ref={ref}
        className={clsx(
          base,
          variants[variant],
          sizes[size],
          round[rounded],
          className,
        )}
        disabled={disabled || loading}
        type="button"
        {...props}
      >
        {loading && <Loader2 className="mr-2 h-4 w-4 animate-spin" />}
        {children}
      </button>
    );
  },
);

Button.displayName = "Button";
