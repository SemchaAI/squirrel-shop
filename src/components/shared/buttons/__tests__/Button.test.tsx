import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "@/components/shared/buttons/Button";

describe("Button component", () => {
  it("renders with default props", () => {
    render(<Button>Click me</Button>);
    const button = screen.getByRole("button", { name: /click me/i });
    expect(button).toBeInTheDocument();
    expect(button).not.toBeDisabled();
    expect(button).toHaveClass("bg-primary");
    expect(button).toHaveClass("text-base");
    expect(button).toHaveClass("rounded-md");
  });

  it("shows loading spinner when loading is true", () => {
    const { container } = render(<Button loading>Loading</Button>);
    expect(screen.getByRole("button")).toBeDisabled();
    const svgEl = container.querySelector("svg.lucide-loader-circle"); //Loader2 Loader2Icon  class
    expect(svgEl).toBeInTheDocument();
  });

  it("is disabled when disabled is true", () => {
    render(<Button disabled>Disabled</Button>);
    expect(screen.getByRole("button")).toBeDisabled();
  });

  it("calls onClick when clicked", () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click</Button>);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("does not call onClick when disabled", () => {
    const handleClick = jest.fn();
    render(
      <Button disabled onClick={handleClick}>
        Can`t click
      </Button>,
    );
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).not.toHaveBeenCalled();
  });
});

const variants = ["primary", "ghost", "text", "destructive"] as const;
const sizes = ["none", "sm", "md", "lg"] as const;
const roundings = ["full", "md", "none"] as const;

describe("Button snapshot combinations", () => {
  it.each(
    variants.flatMap((variant) =>
      sizes.flatMap((size) =>
        roundings.map((rounded) => [variant, size, rounded] as const),
      ),
    ),
  )(
    "matches snapshot for variant='%s', size='%s', rounded='%s'",
    (variant, size, rounded) => {
      const { container } = render(
        <Button variant={variant} size={size} rounded={rounded}>
          {variant} {size} {rounded}
        </Button>,
      );
      expect(container.firstChild).toMatchSnapshot();
    },
  );
});

// it.each([
//   ["sm", ["px-3", "py-1.5", "text-sm"]],
//   ["md", ["px-4", "py-2", "text-base"]],
//   ["lg", ["px-6", "py-3", "text-lg"]],
//   ["none", ["text-sm"]],
// ])("applies size variant '%s'", (size, expectedClasses) => {
//   render(<Button size={size as "sm" | "md" | "lg" | "none"}>{size}</Button>);
//   expect(screen.getByRole("button")).toHaveClass(...expectedClasses);
// });

// it.each([
//   ["md", "rounded-md"],
//   ["full", "rounded-full"],
//   ["none", "rounded-none"],
// ])("applies rounded variant '%s'", (rounded, expectedClass) => {
//   render(
//     <Button rounded={rounded as "md" | "full" | "none"}>Rounded</Button>,
//   );
//   const button = screen.getByRole("button");
//   expect(button).toHaveClass(expectedClass);
// });

// it.each([
//   ["primary", ["bg-primary"]],
//   ["ghost", ["bg-transparent", "text-primary", "border-primary"]],
//   ["text", ["border-transparent", "text-primary", "border-transparent"]],
//   ["destructive", ["bg-error", "border-error", "text-white"]],
// ])("applies visual variant '%s'", (variant, expectedClasses) => {
//   render(
//     <Button variant={variant as "primary" | "ghost" | "text" | "destructive"}>
//       Variant
//     </Button>,
//   );
//   const button = screen.getByRole("button");
//   expect(button).toHaveClass(...expectedClasses);
// });
