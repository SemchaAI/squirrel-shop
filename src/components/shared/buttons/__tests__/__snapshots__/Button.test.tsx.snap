// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='lg', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-6 py-3 text-lg rounded-full"
  type="button"
>
  destructive
   
  lg
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='lg', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-6 py-3 text-lg rounded-md"
  type="button"
>
  destructive
   
  lg
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='lg', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-6 py-3 text-lg rounded-none"
  type="button"
>
  destructive
   
  lg
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='md', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-4 py-2 text-base rounded-full"
  type="button"
>
  destructive
   
  md
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='md', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-4 py-2 text-base rounded-md"
  type="button"
>
  destructive
   
  md
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='md', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-4 py-2 text-base rounded-none"
  type="button"
>
  destructive
   
  md
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='none', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover text-sm rounded-full"
  type="button"
>
  destructive
   
  none
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='none', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover text-sm rounded-md"
  type="button"
>
  destructive
   
  none
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='none', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover text-sm rounded-none"
  type="button"
>
  destructive
   
  none
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='sm', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-3 py-1.5 text-sm rounded-full"
  type="button"
>
  destructive
   
  sm
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='sm', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-3 py-1.5 text-sm rounded-md"
  type="button"
>
  destructive
   
  sm
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='destructive', size='sm', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-error border-error text-white hover:bg-error-hover px-3 py-1.5 text-sm rounded-none"
  type="button"
>
  destructive
   
  sm
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='lg', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-6 py-3 text-lg rounded-full"
  type="button"
>
  ghost
   
  lg
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='lg', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-6 py-3 text-lg rounded-md"
  type="button"
>
  ghost
   
  lg
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='lg', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-6 py-3 text-lg rounded-none"
  type="button"
>
  ghost
   
  lg
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='md', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-4 py-2 text-base rounded-full"
  type="button"
>
  ghost
   
  md
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='md', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-4 py-2 text-base rounded-md"
  type="button"
>
  ghost
   
  md
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='md', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-4 py-2 text-base rounded-none"
  type="button"
>
  ghost
   
  md
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='none', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary text-sm rounded-full"
  type="button"
>
  ghost
   
  none
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='none', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary text-sm rounded-md"
  type="button"
>
  ghost
   
  none
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='none', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary text-sm rounded-none"
  type="button"
>
  ghost
   
  none
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='sm', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-3 py-1.5 text-sm rounded-full"
  type="button"
>
  ghost
   
  sm
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='sm', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-3 py-1.5 text-sm rounded-md"
  type="button"
>
  ghost
   
  sm
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='ghost', size='sm', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-primary text-primary hover:bg-primary hover:text-on-primary px-3 py-1.5 text-sm rounded-none"
  type="button"
>
  ghost
   
  sm
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='lg', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-6 py-3 text-lg rounded-full"
  type="button"
>
  primary
   
  lg
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='lg', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-6 py-3 text-lg rounded-md"
  type="button"
>
  primary
   
  lg
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='lg', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-6 py-3 text-lg rounded-none"
  type="button"
>
  primary
   
  lg
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='md', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-4 py-2 text-base rounded-full"
  type="button"
>
  primary
   
  md
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='md', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-4 py-2 text-base rounded-md"
  type="button"
>
  primary
   
  md
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='md', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-4 py-2 text-base rounded-none"
  type="button"
>
  primary
   
  md
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='none', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover text-sm rounded-full"
  type="button"
>
  primary
   
  none
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='none', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover text-sm rounded-md"
  type="button"
>
  primary
   
  none
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='none', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover text-sm rounded-none"
  type="button"
>
  primary
   
  none
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='sm', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-3 py-1.5 text-sm rounded-full"
  type="button"
>
  primary
   
  sm
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='sm', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-3 py-1.5 text-sm rounded-md"
  type="button"
>
  primary
   
  sm
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='primary', size='sm', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-primary border-primary text-on-primary hover:bg-primary-hover px-3 py-1.5 text-sm rounded-none"
  type="button"
>
  primary
   
  sm
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='lg', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-6 py-3 text-lg rounded-full"
  type="button"
>
  text
   
  lg
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='lg', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-6 py-3 text-lg rounded-md"
  type="button"
>
  text
   
  lg
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='lg', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-6 py-3 text-lg rounded-none"
  type="button"
>
  text
   
  lg
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='md', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-4 py-2 text-base rounded-full"
  type="button"
>
  text
   
  md
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='md', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-4 py-2 text-base rounded-md"
  type="button"
>
  text
   
  md
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='md', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-4 py-2 text-base rounded-none"
  type="button"
>
  text
   
  md
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='none', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline text-sm rounded-full"
  type="button"
>
  text
   
  none
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='none', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline text-sm rounded-md"
  type="button"
>
  text
   
  none
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='none', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline text-sm rounded-none"
  type="button"
>
  text
   
  none
   
  none
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='sm', rounded='full' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-3 py-1.5 text-sm rounded-full"
  type="button"
>
  text
   
  sm
   
  full
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='sm', rounded='md' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-3 py-1.5 text-sm rounded-md"
  type="button"
>
  text
   
  sm
   
  md
</button>
`;

exports[`Button snapshot combinations matches snapshot for variant='text', size='sm', rounded='none' 1`] = `
<button
  class="inline-flex cursor-pointer items-center justify-center border  font-medium transition-colors focus:outline-none disabled:opacity-50 disabled:pointer-events-none bg-transparent border-transparent text-primary hover:underline px-3 py-1.5 text-sm rounded-none"
  type="button"
>
  text
   
  sm
   
  none
</button>
`;
